### PR TITLE
Payment with stripe#187570888

### PR DESCRIPTION
**Pull request by Ishimwe7**
###What does this PR do?
This PR implements payment processing functionality using Stripe for our e-commerce application. It includes the integration of Stripe to handle payment intents, secure payment processing, and confirmation of payments from the frontend to the backend.

###Description of Task to be completed?
Install and configure Stripe library in the backend.
Create an endpoint to generate payment intents.
Handle payment confirmation and error handling on the frontend.
Add documentation and example usage for the payment processing feature.

###How should this be manually tested?
Clone the repository and checkout to this PR's branch.
Install necessary dependencies using npm install.
Start the server using npm start.
Use Postman or Swagger doc to test the "api/payment/pay" endpoint by sending a POST request with a JSON body containing the amount and currency.
Verify that a payment intent is created and a client secret is returned.

###What are the relevant pivotal tracker stories
https://www.pivotaltracker.com/story/show/187570888

### CODE REVIEW

**Comment by rodriguecyber** [comment]
Screenshot 2024-06-07 at 19 12 51
 correct this

**Comment by rodriguecyber** [comment]
and `userId`  should not being in request body should  be got in token

File: src/controllers/paymentController/paymentController.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        return res.status(500).json({ message: error.message });
+      }
+    }
+  }
+
+  async paymentCancel(req: Request, res: Response) {
+    const userId = req.body.userId as string;
```
Comment (kirengamartial) [inline src/controllers/paymentController/paymentController.ts:110]: userId should not be from the req.body and it's better that it has the data type of number not string

**Review by kirengamartial — CHANGES_REQUESTED** [review]
consider the review

**Review by kirengamartial — APPROVED** [review]
consider the changes requested

File: src/models/paymentModel.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  id: number;
+  stripeId: string;
+  userId: number;
+  orderId: number;
+  amount: number;
+  currency: string;
+  payment_status: PaymentStatus;
+  payment_method: PaymentMethod;
+  createdAt?: Date;
+  updatedAt?: Date;
```
Comment (derrick-nuby) [inline src/models/paymentModel.ts:29]: since the payment is one time I do not think we need updated at field

**Review by derrick-nuby — APPROVED** [review]
Looks good and ready for merge if feedback is considered

**Review by DeliceLydia — APPROVED** [review]

**Review by Balinda21 — COMMENTED** [review]
Great work @Ishimwe7

**Pull request merged**